adobe/spectrum-css
# fix: restore full builds for legacy components & disable some lint-staged checks

## Description

* Requires the legacy `component-builder` to use newly created `legacy-*` PostCSS plugins which were restored from [the commit prior](https://github.com/adobe/spectrum-css/commit/bc6c1348e93698bfe3a560a2b147fb9f4b964de5) to our change to allow pass-throughs.
* Fixes `remapvars` functions with broken class name regex usage due to Prettier modifying them & marks them with `prettier-ignore` comments to _hopefully_ prevent this in the future.
* Removes CSS files from Prettier's `lint-staged` check.
* Removes Stylelint from the `lint-staged` check in favor of handling it via Pull Requests & GH Actions.
* Updates the Stylelint config to remove the `max-nesting-depth` check
* Configures Stylelint to ignore the "new" `legacy-*` plugins

This ALSO fixes an issue where non-migrated component doc pages were broken on Netlify deploys. Our GitHub Action for building & deploying was faulty.
* During the build step, while running the task to build Storybook, Nx was _not_ recognizing the cache for the `build:components` task, and was re-building all of the components. For some reason, during the HTML docs compilation, this was creating invalid asset paths for non-migrated components and the pages would show as broken on any Netlify deploy.

## Steps to Validate
* Pull this branch down locally
* Run `git clean -dfX`
* Run `yarn install`
* Run `yarn cache:clean`
* Run `yarn clean`
* Run `yarn build:all` and observe that the components, docs site, and Storybook all build successfully.
* Open any non-migrated component's directory (currently Tabs, Table Accordion are all good candidates), and look for the `dist` directory. In there, open the `vars.css` file and observe that CSS is actually in there. When broken, this file exists but is empty.

Additionally:
* Run `yarn dev` to start the docs site locally, and click through the site to be sure that all of the pages are working as expected.
* Also please visit the deploy URL for this PR and be sure that both migrated and non-migrated component pages are rending in the docs site and that the switchers work, etc.
* Also please visit the storybook deploy to be sure that Storybook is working correctly.

## Screenshots



## To-do list



- [ ] I have updated any relevant storybook stories and templates.
- [ ] If my change(s) include visual change(s), a designer has reviewed and approved those changes.
